data/src/test/java/com/linkedin/data/schema/validation/TestValidation.java

```diff
@@ -2081,4 +2098,161 @@ public void testNonRootStartDataElement() throws IOException
       }
     }
   }
+
+  @Test
+  public void testUnrecognizedFieldValidation() throws IOException
+  {
+    String schemaText =
+        "{ \"type\" : \"record\", \"name\" : \"foo\", \"fields\" : " +
+            "[ { \"name\" : \"bar\", \"type\" : \"string\" } ] }";
+
+    Object goodObjects[] =
+        {
+            "a valid string"
+        };
+
+    Object badObjects[] =
+        {
+        };
+
+    // There is no coercion for this type.
+    // Test with all coercion modes, result should be the same for all cases.
+    testCoercionValidation(schemaText, "bar", goodObjects, badObjects, disallowUnrecognizedFieldOption());
+    testCoercionValidation(schemaText, "bar", goodObjects, badObjects, trimUnrecognizedFieldOption());
```

Comment:
This test, as far as I can tell, will assert that the result of the validation is false (meaning it failed). Why would such validation fail? If we trim, doesn't that lead to successful validation?